### PR TITLE
TECH-443 Wordpress 6.1 Compatibility

### LINKED ISSUE #1: wrong link in readme
In the Readme at the bottom you have

a link where it says: "For a full list of of available options please the setup guide: Online Bookings with Wordpress and Checkfront"

The link is pointing to setup instructions for Joomla and not WordPress.

### LINKED ISSUE #2: Update to latest WP SVN commit
Hi there, great that you have also added this project to github. Could you update the github codebase to the latest version hosted in the WordPress SVN?

### PR DESCRIPTION
**Pull request by Nebual**
JIRA: [TECH-443](https://checkfront.atlassian.net/browse/TECH-443)

* Sync Git version to latest Wordpress Plugin SVN state
* Remove Sidebar Widget, its been broken for a long time
* Improve validation/escaping
* use wp_enqueue_scripts instead of echo'ing raw <script> tags
* remove Facebook/Google+, as, in the legendary words of Wordpress Developer Support:

> Facebook is extremely problematic as they track pretty much everyone and everything.

I recommend reviewing by-commit, and ignore the first commit, that's a raw resync from the SVN (live) state :see_no_evil: 

Fixes #1
Fixes #2

### CODE REVIEW

**Comment by Nebual** [comment]
This can be tested following the instructions in [Wordpress Plugin Testing](https://checkfront.atlassian.net/wiki/spaces/QT/pages/1944649763/WordPress+Plugin+Testing), using the following zip:

[checkfront-wp-booking.zip](https://github.com/Checkfront/Wordpress-Plugin/files/9924875/checkfront-wp-booking.zip)

File: CheckfrontWidget.php [hunk 1]
```diff
@@ -113,9 +108,6 @@ function set_pipe($url) {
 	private function set_host($host) {
 		$this->host = $host;
 		$this->url = "//{$this->host}";
-
-		// v1 only
-		$this->api_url = "{$this->url}/api/2.0";
```
Comment (YourFavouriteNick) [inline CheckfrontWidget.php:118]: is nothing using this anymore?

File: CheckfrontWidget.php [hunk 2]
```diff
@@ -210,29 +174,40 @@ private function set_ids($ids) {
...
 
     /**
-     * render v2 interface
+     * render v2 booking portal
      *
      * @param array $cnf shortcode paramaters
      * @return string $html rendering code
     */
-	private function v2_interface($arg=array()) {
+	private function portal($arg=array()) {
```
Comment (YourFavouriteNick) [inline CheckfrontWidget.php:182]: I love the better naming

File: CheckfrontWidget.php [hunk 3]
```diff
@@ -210,29 +174,40 @@ private function set_ids($ids) {
...
-			'widget_id'=>'',
-			'item_id'=>'',
-			'category_id'=>'',
-			'theme'=>'',
-			'layout'=>'',
-			'tid'=>'',
-			'options'=>'',
-			'date'=>'',
-			'style'=>''
+			'filter_category_id' => '',
```
Comment (YourFavouriteNick) [inline CheckfrontWidget.php:185]: should we mention the new options in the readme? good eye finding these

File: checkfront.php [hunk 4]
```diff
@@ -15,209 +16,132 @@
...
+	add_filter('comments_template', 'checkfront_comments_template_filter', 10, 1);
+
+	// disable auto p
+	// remove_filter ('the_content', 'wpautop');
+
+	// disable wptexturize
+	remove_filter('the_content', 'wptexturize');
 }
 
 // disable comments on booking pagfe
```
Comment (YourFavouriteNick) [inline checkfront.php]: could you fix that typo?

**Review by YourFavouriteNick — APPROVED** [review]

File: checkfront.php [hunk 5]
```diff
@@ -71,51 +71,26 @@ function checkfront_setup()
...
-			$Checkfront->arg['widget'] = 0;
-		} else {
-			if (is_page() && !$checkfront_widget_page) {
-				$Checkfront->arg['widget'] = 0;
-			}
-			if (is_single() && !$checkfront_widget_post) {
-				$Checkfront->arg['widget'] = 0;
-			}
-		}
+	if (stripos($post->post_content, '[checkfront') === false) {
```
Comment (YourFavouriteNick) [inline checkfront.php:78]: thank you, this is much clearer

File: setup.php [hunk 6]
```diff
@@ -10,25 +10,6 @@
...
 <div style="width: 800px">
-	<script type="text/javascript">
-		jQuery(document).ready(function() {
-
-			function clean(str) {
-				return str.replace(/[^\d\w\-\_ , "]/ig, '');
-			}
-
-			jQuery('#shortcode_generator').click(function() {
-				height = 600;
```
Comment (YourFavouriteNick) [inline setup.php]: is this shortcode generator gone?

**Review by YourFavouriteNick — APPROVED** [review]

**Review by brettgoss — APPROVED** [review]
Overall looks good to me. Would love to see some code style enforced, but glad that you're at least incrementally making it better even if it's still inconsistent.

File: setup.php (hunk 6 above)
Comment (Nebual) [reply setup.php]: I think the js wasn't needed anymore, as the current live version of the generator just links with an
```
<a
  class="button-primary"
  href="https://<?php echo trim($Checkfront->host) ?>/manage/extend/integrate/droplet/?src=wordpress"
  value=" Update "
  target="_blank"
  id="shortcode_generator"
/>Launch Shortcode Generator</a>
```
where the `_blank` means "open in new tab" in modern browsers. I think the js was trying to open it in a new window of a specified size, but browsers haven't really allowed that in awhile.

**Comment by icomeaucheckfront** [comment]
In the README.md file line 10 we state `This plugin is for Wordpress 1.6 or greater (tested to 2.5). ` 

Should we update this part to WordPress v6.1?

**Comment by brettgoss** [comment]
@icomeaucheckfront good catch. The readme.md in general is quite outdated. It's not used anywhere besides this repo, but should probably be updated nonetheless (and could close #1 while you're at it)

**Comment by icomeaucheckfront** [comment]
I'm not familiar with using this GitHub to comment on specific lines within specific files but in the file 
setup.php:21 - Could we change the wording of 'short code' to 'shortcode' in the href ?

File: checkfront.php [hunk 7]
```diff
@@ -81,7 +79,7 @@ function checkfront_head()
 		return;
 	}
 
-	echo ' <script src="//' . $Checkfront->host . '/lib/interface--' . $Checkfront->interface_version . '.js" type="text/javascript"></script>' ."\n";
+	wp_enqueue_script('cf/interface.js', "//{$Checkfront->host}/lib/interface--{$Checkfront->interface_version}.js", ['jquery']);
```
Comment (alanbrown-ca) [inline checkfront.php:82]: Excellent

**Review by alanbrown-ca — APPROVED** [review]

File: setup.php [hunk 8]
```diff
@@ -1,193 +1,127 @@
...
+			<li style="padding: 5px 0; height: 25px; line-height: 25px;"><strong style="margin-right: 10px; background-color: #ccc; border-radius: 20px; clear: none; color: #000; display: inline-block; float: left; font-style: normal; height: 25px; font-weight: bold; line-height: 25px;font-size:14px; text-align: center; width: 25px;">2</strong> Supply your Checkfront URL and optional settings <a href="#checkfront_setup">below</a>.</li>
+			<li style="padding: 5px 0; height: 25px; line-height: 25px;"><strong style="margin-right: 10px; background-color: #ccc; border-radius: 20px; clear: none; color: #000; display: inline-block; float: left; font-style: normal; height: 25px; font-weight: bold; line-height: 25px;font-size:14px; text-align: center; width: 25px;">3</strong> Create a new page and supply the <a href="#shortcode">short code</a> you created.</li>
```
Comment (icomeaucheckfront) [inline setup.php]: Could we change the spelling of the href to 'shortcode' instead of 'short code'

File: CheckfrontWidget.php (hunk 1 above)
Comment (Nebual) [reply CheckfrontWidget.php:118]: Long dead code (thats from the "svn sync" commit)

**Comment by Nebual** [comment]
> The readme.md in general is quite outdated. It's not used anywhere besides this repo

I think I'll just delete the README.md in favour of the readme.txt wordpress needs.

**Pull request merged**